packages/ui/useScroll/Scrollbar.js

```diff
@@ -33,87 +34,120 @@ const setCSSVariable = (element, variable, value) => {
+const useScrollbar = (props) => {
+	const {className, clientSize, corner, css, minThumbSize, vertical, ...rest} = props;
 	// Refs
 	const uiScrollbarContainerRef = useRef();
-	const thumbRef = useRef();
-	const hideThumbJob = useRef(null);
-	// Render
-	const
-		{childRenderer, className, corner, css, minThumbSize, vertical, ...rest} = props,
-		containerClassName = classNames(
-			className,
-			corner && css.corner,
-			css.scrollbar,
-			vertical ? css.vertical : css.horizontal
-		);
-
-	delete rest.clientSize;
-
-	hideThumbJob.current = hideThumbJob.current || new Job(hideThumb, thumbHidingDelay);
-
-	function hideThumb () {
-		removeClass(thumbRef.current, css.thumbShown);
+	const scrollbarTrackRef = useRef();
+	const hideScrollbarTrackJob = useRef(null);
+
+	hideScrollbarTrackJob.current = hideScrollbarTrackJob.current || new Job(hideScrollbarTrack, scrollbarTrackHidingDelay);
+
+	function hideScrollbarTrack () {
+		removeClass(scrollbarTrackRef.current, css.scrollbarTrackShown);
 	}
 
 	useEffect(() => {
 		return () => {
-			hideThumbJob.current.stop();
+			hideScrollbarTrackJob.current.stop();
 		};
 	}, []);
 
-	useImperativeHandle(ref, () => ({
-		getContainerRef: () => (uiScrollbarContainerRef),
-		showThumb: () => {
-			hideThumbJob.current.stop();
-			addClass(thumbRef.current, css.thumbShown);
+	function getContainerRef () {
+		return uiScrollbarContainerRef;
+	}
+
+	function showScrollbarTrack () {
+		hideScrollbarTrackJob.current.stop();
+		addClass(scrollbarTrackRef.current, css.scrollbarTrackShown);
+	}
+
+	function startHidingScrollbarTrack () {
+		hideScrollbarTrackJob.current.start();
+	}
+
+	function update (bounds) {
+		const
+			primaryDimenstion = vertical ? 'clientHeight' : 'clientWidth',
+			trackSize = clientSize ? clientSize[primaryDimenstion] : uiScrollbarContainerRef.current[primaryDimenstion],
+			scrollViewSize = vertical ? bounds.clientHeight : bounds.clientWidth,
+			scrollContentSize = vertical ? bounds.scrollHeight : bounds.scrollWidth,
+			scrollOrigin = vertical ? bounds.scrollTop : bounds.scrollLeft,
+			scrollbarThumbSizeRatioBase = (scrollViewSize / scrollContentSize),
+			scrollbarThumbProgressRatio = (scrollOrigin / (scrollContentSize - scrollViewSize)),
+			scrollbarThumbSizeRatio = Math.max(ri.scale(minThumbSize) / trackSize, Math.min(1, scrollbarThumbSizeRatioBase));
+
+		setCSSVariable(scrollbarTrackRef.current, '--scrollbar-thumb-size-ratio', scrollbarThumbSizeRatio);
+		setCSSVariable(scrollbarTrackRef.current, '--scrollbar-thumb-progress-ratio', scrollbarThumbProgressRatio);
+	}
+
+	return {
+		imperativeHandles: {
+			getContainerRef,
+			showScrollbarTrack,
+			startHidingScrollbarTrack,
+			update
 		},
-		startHidingThumb: () => {
-			hideThumbJob.current.start();
+		restProps: rest,
+		scrollbarProps: {
+			className: classNames(
+				className,
+				{[corner]: css.corner},
+				css.scrollbar,
+				vertical ? css.vertical : css.horizontal
+			),
+			ref: uiScrollbarContainerRef
```

Comment:
I don't think that we need the `ui` prefix for this `ref` as we did with `scrollContainerRef`. For `theme` refs, we could add a `theme` word like `useThemeScroll`.

Reply:
👍